specification/cognitiveservices/Language.Conversations/main.tsp

```diff
@@ -0,0 +1,1841 @@
+}
+
+@doc("The input for a custom conversation task.")
+model ConversationalTask extends AnalyzeConversationTask {
+  @doc("The input for a custom conversation task.")
+  kind: AnalyzeConversationTaskKind.Conversation;
+
+  @doc("The input ConversationItem and its optional parameters")
+  analysisInput: ConversationAnalysisOptions;
+
+  @doc("Input parameters necessary for a Conversation task.")
+  parameters: ConversationTaskParameters;
+}
+
+@doc("The input ConversationItem and its optional parameters")
+model ConversationAnalysisOptions {
+  #suppress "@azure-tools/typespec-azure-core/bad-record-type"
+  @doc("The abstract base for a user input formatted conversation (e.g., Text, Transcript).")
+  conversationItem: ConversationItemBase;
+}
+
+#suppress "@azure-tools/typespec-azure-core/bad-record-type"
+@doc("The abstract base for a user input formatted conversation (e.g., Text, Transcript).")
+model ConversationItemBase is Record<unknown> {
+  @doc("The ID of a conversation item.")
+  id: string;
+
+  @doc("The participant ID of a conversation item.")
+  participantId: string;
+
+  @doc("The override language of a conversation item in BCP 47 language representation.")
+  language?: string;
+
+  @doc("Enumeration of supported conversational modalities.")
+  modality?: InputModality;
+
+  @added(Versions.v2023_04_01)
+  @doc("Role of the participant.")
+  role?: role;
+}
+
+@added(Versions.v2023_04_01)
+@doc("Role of the participant.")
+union role {
+  string,
+
+  @added(Versions.v2023_04_01)
+  @doc("The participant is a customer.")
+  customer: "customer",
+
+  @added(Versions.v2023_04_01)
+  @doc("The participant is an agent.")
+  agent: "agent",
+
+  @added(Versions.v2023_04_01)
+  @doc("The is a generic participant.")
+  generic: "generic",
+}
+
+#suppress "@azure-tools/typespec-azure-core/bad-record-type"
+@doc("The text modality of an input conversation.")
+model TextConversationItem is ConversationItemBase {
+  @doc("The text input")
+  text: string;
+}
+
+@doc("This is the parameter set of either the Orchestration project itself or one of the target services.")
+@discriminator("targetProjectKind")
+model AnalysisParameters {
+  @doc("The type of a target service")
+  targetProjectKind: targetProjectKind;
+
+  @doc("The API version to use when call a specific target service.")
+  apiVersion?: string;
+}
+
+@doc("A wrap up of non-linked intent response.")
+model NoneLinkedTargetIntentResult extends TargetIntentResult {
+  @doc("The actual response from a Conversation project.")
+  targetProjectKind: targetProjectKind.NonLinked;
+
+  @doc("The actual response from a Conversation project.")
+  result?: ConversationResult;
+}
+#suppress "@azure-tools/typespec-azure-core/bad-record-type" "LuisParameters should not use 'extends Record<unknown>'. Use 'extends Record<string>' instead."
+@doc("This is a set of request parameters for LUIS Generally Available projects.")
+model LuisParameters extends AnalysisParameters {
+  ...Record<unknown>;
```

Comment:
What is the intent behind spreading Record<unknown> here? What service behavior are we trying to convey? I suspect that this is not correct.

Reply:
Looks like a bug and can be removed

Reply:
This seems incorrect as well. Removing.

Reply:
But this might trigger breaking change as we already have the additionalProperties set to true here.

Reply:
I can help navigate the breaking change process. Let's make sure we set ourselves up for the brightest future that we can.